llvm/lib/Transforms/Scalar/GVN.cpp

```diff
@@ -3280,7 +3280,7 @@ class llvm::gvn::GVNLegacyPass : public FunctionPass {
     if (skipFunction(F))
       return false;
 
-    auto *LIWP = getAnalysisIfAvailable<LoopInfoWrapperPass>();
+    auto &LIWP = getAnalysis<LoopInfoWrapperPass>();
```

Comment:
I prefer not to leave it as is - I think either we should get the analysis with `getAnalysis` or we should remove the `addRequired`, so that they are consistent with each other.